qfieldsync/core/cloud_transferrer.py

```diff
@@ -623,7 +717,9 @@ def __init__(
                 self.cloud_project,
                 self.transfer_type,
                 file,
-                self.temp_dir.joinpath(str(self.transfer_type.value), file.name),
+                Path(file.local_dir).joinpath(file.name)
+                if use_file_local_dir
+                else self.temp_dir.joinpath(str(self.transfer_type.value), file.name),
             )
```

Comment:
Can we move this to a variable, it's unreadable as a parameter?

Also, what if `temp_dir` is `None`?

Reply:
Fixed, it'll never be None (like it used to be)